tools/ovc/openvino/tools/ovc/convert_impl.py

```diff
@@ -0,0 +1,831 @@
+        # corresponds to TF1 If and TF1 While operations
+        "Switch", "Merge",
+        # corresponds to operations with complex tensors
+        "FFT", "FFT2D", "FFT3D", "IFFT", "IFFT2D", "IFFT3D",
+        "RFFT", "RFFT2D", "RFFT3D", "IRFFT", "IRFFT2D", "IRFFT3D",
+        "Complex", "ComplexAbs", "Real", "Imag",
+    ]
+    if len(conversion_error_match) < 1 or len(conversion_error_match[0]) != 4:
+        # no match for the fallback by unsupported operation
+        return False
+
+    unsupported_operations = conversion_error_match[0][1].replace(" ", "").split(",")
+    fallback_operations = [operation for operation in unsupported_operations if operation in all_fallback_operations]
+
+    if len(fallback_operations) == 0:
+        return False
+
+    fallback_reasons.append("Fallback to the legacy TF FE due to operation(s): " + ', '.join(fallback_operations))
+    return True
+
+
+def get_default_frontends():
+    # Set which frontend to use by default, values should be 'new' or 'legacy'
+    default_frontends = {
+        'onnx': 'new',
+        'tf': 'new'
+    }
+    return default_frontends
+
+
+def get_moc_frontends(argv: argparse.Namespace):
+    fem = argv.feManager
+
+    # Read user flags:
+    use_legacy_frontend = argv.use_legacy_frontend
+    use_new_frontend = argv.use_new_frontend
+
+    if not fem or use_legacy_frontend:
+        return None, []
+
+    available_moc_front_ends = get_available_front_ends(fem)
+
+    if not argv.framework and argv.input_model:
+        moc_front_end = fem.load_by_model(argv.input_model)
+        if not moc_front_end:
+            return None, available_moc_front_ends
+        argv.framework = moc_front_end.get_name()
+    elif argv.framework in available_moc_front_ends:
+        moc_front_end = fem.load_by_framework(argv.framework)
+    else:
+        return None, []
+
+    default_frontends = get_default_frontends()
+    # Disable MOC frontend if default is set to legacy and no user override
+    if default_frontends.get(moc_front_end.get_name()) == 'legacy' and not use_new_frontend:
+        return None, available_moc_front_ends
+
+    # This check as a workaround to skip IR frontend
+    if not moc_front_end.get_name() in available_moc_front_ends:
+        return None, available_moc_front_ends
+
+    return moc_front_end, available_moc_front_ends
+
+
+def prepare_ir(argv: argparse.Namespace):
+    # TODO: remove this workaround once new TensorFlow frontend supports non-frozen formats: checkpoint, MetaGraph, and SavedModel
+    # Now it converts all TensorFlow formats to the frozen .pb format in case new TensorFlow frontend
+    is_tf, _, _, _, _ = deduce_legacy_frontend_by_namespace(argv)
+    argv = arguments_post_parsing(argv)
+    t = tm.Telemetry()
+
+    graph = None
+    fallback_reasons = []
+    moc_front_end, available_moc_front_ends = get_moc_frontends(argv)
+    if moc_front_end:
+        fallback_reasons = check_fallback(argv)
+        if len(fallback_reasons) == 0:
```

Comment:
please create a ticket and mention it in todo. It is important because I do not know what will be in case of `ovc` and fallback.